code/modules/modular_computers/file_system/programs/paperworkprinter.dm

```diff
@@ -0,0 +1,75 @@
+//TODO: add more sane printing handling
+/datum/computer_file/program/paperwork_printer
+	filename = "ppwrkprnt"
+	filedesc = "Paperwork Printing"
+	category = PROGRAM_CATEGORY_MISC
+	program_icon_state = "id"
+	extended_desc = "Program for dispensing paperwork"
+	requires_ntnet = FALSE
+	size = 4
+	tgui_id = "NtosPaperworkPrinter"
+	program_icon = "clipboard-list"
+	var/obj/item/paper/paperwork/which
+	var/print = 1
+
+/datum/computer_file/program/paperwork_printer/ui_static_data(mob/user)
+	var/list/data = get_header_data()
+	var/obj/item/computer_hardware/printer/printer
+	if(computer)
+		printer = computer.all_components[MC_PRINT]
+
+	if(computer)
+		data["have_printer"] = !!printer
+	else
+		data["have_printer"] = FALSE
+	return data
+
+/datum/computer_file/program/paperwork_printer/ui_act(action, params, datum/tgui/ui)
+	if(..())
+		return
+
+	switch(action)
+		if("PRG_print_GenReqForm")
+			which = /obj/item/paper/paperwork/general_request_form
```

Comment:
i would have the action be "PRG_print_form" or something similar and decide which paper to use using params instead of a unique action for each paper type